pkg/cluster/cluster_config.go

```diff
@@ -21,6 +22,69 @@ import (
 
 // +kubebuilder:rbac:groups="config.openshift.io",resources=ingresses,verbs=get
 
+type Platform string
+
+// Release includes information on operator version and platform
+// +kubebuilder:object:generate=true
+type Release struct {
+	Name    Platform                `json:"name,omitempty"`
+	Version version.OperatorVersion `json:"version,omitempty"`
+}
+
+var clusterConfig struct {
+	Namespace string
+	Platform  Platform
+	Release   Release
+}
+
+// Init initializes cluster configuration variables on startup
+// init() won't work since it is needed to check the error.
+func Init(ctx context.Context, cli client.Client, log logr.Logger) error {
+	var err error
+
+	clusterConfig.Namespace, err = getOperatorNamespace()
```

Comment:
in configureAlertManager and in watchMonitoringSecretResource the error causes return. In other places where the value compared "" does not change the flow. I did not revisit callees' logic. It may make sense.

> should this not be fatal => os.Exit(1) ?

it's not fatal. Do you mean it should be?

Reply:
i am thinking it should be fatal.
for this comments `// not fatal, fallback to ""`

Reply:
> i am thinking it should be fatal. for this comments `// not fatal, fallback to ""`

It changes behaviour. Mean, we continued to work if no environment variable or cannot read namespace from the filesystem, but if it's fatal we don't.

On the other hand with the environment variable corner cases (like tests) are covered.

Reply:
i understand.
1. env
2. check /var/...
3. log as error but continue
but the consequence on (3) will leave to later problem, so if it should os.Exit(1) directly in main process or not.
e.g i can understand someone build the whole product without having (1) set, but the reason can skip (2) is something really wrong with the pod. 

but this is a question out of the scope of your PR.
the "" has been there, we do not need to change anything here